pkg/controller/capacity/deployment.go

```diff
@@ -94,3 +95,62 @@ func (c Controller) calculatePercentageFromAmount(total, amount int32) int32 {
 
 	return int32(math.Ceil(result))
 }
+
+type sadContainerSummary struct {
+	pods    int
+	reasons map[string]struct{}
+}
+
+func summarizeSadPods(sadPods []shipper.PodStatus) string {
+	sadContainerReasons := make(map[string]*sadContainerSummary)
+
+	for _, sadPod := range sadPods {
+		summarizeContainers(sadPod.InitContainers, sadContainerReasons)
+		summarizeContainers(sadPod.Containers, sadContainerReasons)
+	}
+
+	containers := make([]string, 0, len(sadContainerReasons))
+	for c, _ := range sadContainerReasons {
+		containers = append(containers, c)
+	}
+
+	sort.Strings(containers)
+
+	summaryStrs := make([]string, 0, len(sadContainerReasons))
+	for _, container := range containers {
+		summary := sadContainerReasons[container]
+		reasons := make([]string, 0, len(summary.reasons))
+		for r, _ := range summary.reasons {
+			reasons = append(reasons, r)
+		}
+
+		sort.Strings(reasons)
+
+		summaryStrs = append(summaryStrs, fmt.Sprintf("%dx%q containers with %v", summary.pods, container, reasons))
+	}
+
+	return strings.Join(summaryStrs, "; ")
+}
+
+func summarizeContainers(containers []corev1.ContainerStatus, sadContainerReasons map[string]*sadContainerSummary) {
+	for _, container := range containers {
+		if container.Ready {
+			continue
+		}
+
+		if sadContainerReasons[container.Name] == nil {
```

Comment:
Would you please elaborate what this case means? At first it looked like a miss in the hashtable lookup, but i see it's a nil-check. Where do nils are supposed to come from?

Reply:
It *should* check for a lookup miss. Thanks for spotting!